web/frontend/javascripts/cluster_check_settings.js

```diff
@@ -0,0 +1,215 @@
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    get('/api/checks/catalog').then(({ data }) => {
+      setChecksCatalog(data);
+    });
+    get(`/api/checks/${clusterId}/settings`)
+      .then(({ data }) => {
+        const {
+          hosts,
+          connection_settings: connectionSettings,
+          selected_checks: selectedChecks,
+        } = data;
+        const hostnames = getHostnames(hosts);
+        const newSettings = mergeConnectionSettings(
+          hostnames,
+          connectionSettings
+        );
+        setSettings(newSettings);
+        setSelectedChecks(selectedChecks);
+        setLoading(false);
+      })
+      .catch((error) => {
+        logError(error);
+        setSelectedChecks([]);
+        setSettings({});
+        setLoading(false);
+      });
+  }, [modalOpen]);
+
+  const submit = useCallback(() => {
+    const payload = {
+      selected_checks: selectedChecks,
+      connection_settings: settings,
+    };
+    setLoading(true);
+    post(`/api/checks/${clusterId}/settings`, payload)
+      .then(() => {
+        setLoading(false);
+      })
+      .catch((err) => {
+        logError(err);
+        setLoading(false);
+      });
+  }, [selectedChecks, settings]);
+
+  return (
+    <Fragment>
+      <Button variant="secondary" size="sm" onClick={() => setModalOpen(true)}>
+        <i className="eos-icons eos-18">settings</i>Settings
+      </Button>
+      <Modal show={modalOpen}>
+        <Modal.Header>
+          <Modal.Title>Cluster settings</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <h6>Connection settings</h6>
+          <Accordion>
+            <Card>
+              <Card.Header>
+                Host connection settings
+                <AccordionToggle
+                  className="float-right"
+                  eventKey="connection-settings"
+                />
+              </Card.Header>
+              <Accordion.Collapse eventKey="connection-settings">
+                <Card.Body className="card-check-selection">
+                  <Table>
+                    <thead>
+                      <tr>
+                        <th>Host</th>
+                        <th>Connection user</th>
+                        <th>Default user</th>
+                      </tr>
+                    </thead>
+                    <tbody>
+                      {Object.keys(settings).map((host) => (
+                        <tr key={host}>
+                          <td>{host}</td>
+                          <td>
+                            <Form.Control
+                              size="sm"
+                              value={settings[host]}
+                              onChange={({ target: { value } }) =>
+                                setSettings({ ...settings, [host]: value })
+                              }
+                            />
+                          </td>
+                          <td>root</td>
```

Comment:
This `root` shouldn't come by default. This now comes from the backend

Reply:
You're right but I wanted to wait until we had the hosts projection. Maybe not the greatest choice here, I should have talked with the team before

Reply:
This information comes from the `cloud` discovery loop in any case.

In my opinion the connection settings are something that we could project.
Once we receive the hosts, cluster and cloud data, we create the `ConnectionSettings` entry in the database (this struct already exists in the `check_settings.go` model. We could add the `DefaultUser` column.

This way, you could just consume this table instead of getting the clusters, hosts, etc lists.
The work is moved to the projecto (which actually knows this data), and the remaining services and controllers are much easier.

This is design discussion that we can have live though